core/hydromet_stratified.py

```diff
@@ -383,6 +381,169 @@ def Scenarios_low_and_high_S(df_runoff_SR1: pd.DataFrame, mu: float, GEV_paramet
+    temp_0 = 0.5 - df.sort_values('ratio',ascending=False)['ratio']*.5
+    temp_12 = 0.5
+    temp_24 = 1- temp_0.sort_values(0,ascending=False)
+    raw_df['ratio'] = ""
+    raw_df.loc[0:9,'ratio']= temp_0.values
+    raw_df.loc[9:18,'ratio'] = temp_24.values
+    raw_df.loc[9,'ratio'] = temp_12
+    raw_df['slope_raw'] = raw_df['ratio'].diff() / raw_df['time'].diff()
+    raw_df.loc[0,'slope_raw'] = 0
+    raw_df['slope_dif'] = raw_df.loc[0:9]['slope_raw'].diff()
+
+    df2 = raw_df.set_index(['time'])
+    a = ((2/3)* df2.at[9.0,'ratio']-df2.at[6.0,'ratio'])/18
+    b = (df2.at[6.0,'ratio']-36* a)/6
+    low_12h = 4 * df.loc['24h','value']*(1/36 +2/9 * df.loc['06h','value']/df.loc['24h','value'])
+    up_12h = 2/3 * df.loc['24h','value']*(5/6+2/3 * df.loc['06h','value']/df.loc['24h','value'])
+    ##fix negatives
+    if b < 0:
+        0
+    if a < 0:
+        df2.at[9.0,'ratio']/81
+    ##fix 0 slope
+    if 18*a+b<0:
+        df2.at[9.0,'ratio']/4.5
+    if 18*a+b<0:
+        (-1*b/18)
+    a2 = (9/10.5* df2.at[10.5,'ratio'] - df2.at[9.0,'ratio'])/13.5
+    b2 = (df2.at[9.0,'ratio'] -81 *a2) / 9
+    ##check 2h rainfall
+    up_2 = 2* df.loc['24h','value']*(0.5-(df2.at[11.5,'ratio']+ 3*df2.at[10.5,'ratio'])/4)+.01
+    low_2 = 2* df.loc['24h','value']*(0.5-(3*df2.at[11.5,'ratio']+df2.at[10.5,'ratio'])/4)+.01
+    if df.loc['02h','value']<low_2:
+        test1 = low_2
+    else:
+        test1 = df.loc['02h','value']
+    if df.loc['02h','value']> up_2:
+        test2= up_2
+    else:
+        test2 = df.loc['02h','value']
+    if test1 > test2:
+        test3 = test1
+    else:
+        test3 = test2
+    if test2 > test3:
+        test4 = test2
+    else:
+        test4 = test3
+    if test4>up_2:
+        test_f = up_2
+    else:
+        test_f = test4
+    a3 = 2*(df2.at[11.5,'ratio'] - 2*(0.5-0.5* test_f / df.loc['24h','value'])+ df2.at[10.5,'ratio'])
+    b3 = df2.at[11.5,'ratio']- df2.at[10.5,'ratio'] - 22*a3
+    c3 = (0.5 - 0.5 * test_f /df.loc['24h','value']) - 121* a3 - 11 * b3  
+    ##write ratios
+    ratio_to_24h['ratio'] = 0
+    ratio_to_24h.loc[0:90,'ratio'] = a*np.power(ratio_to_24h.loc[0:90].index/10,2)+b*ratio_to_24h.loc[0:90].index/10
+    ratio_to_24h.loc[91:105,'ratio'] = a2*np.power(ratio_to_24h.loc[91:105].index/10,2)+ b2*ratio_to_24h.loc[91:105].index/10
+    ratio_to_24h.loc[106:115,'ratio'] = a3*np.power(ratio_to_24h.loc[106:115].index/10,2)+ b3*ratio_to_24h.loc[106:115].index/10 + c3
+    ##extra work to get 11.6,11.7
+    ratio_to_24h['slope'] = ratio_to_24h['ratio'].diff()/0.1                                                               
+    if -0.867*ratio_to_24h.loc[115,'slope']+ 0.4337 < 0.399: 
+        fac_116 = -0.867*ratio_to_24h.loc[115,'slope']+ 0.4337
+    else:
+        fac_116 = 0.399
+    if -0.4917*ratio_to_24h.loc[115,'slope']+ 0.8182 < 0.799: 
+        fac_117 = -0.4917*ratio_to_24h.loc[115,'slope']+ 0.8182
+    else:
+        fac_116 = 0.799
```

Comment:
I think this is supposed to be fac_117

Reply:
Check the excel to make sure fac_116 should be changed to fac_117.

Reply:
Checked